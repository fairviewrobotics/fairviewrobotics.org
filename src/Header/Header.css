.nav-custom {
  margin-bottom: 0;
  border-bottom: 1px solid rgba(0,0,0,.3);
  text-transform: uppercase;
  font-family: Montserrat, "Helvetica Neue", Helvetica, Arial, sans-serif;
  background-color: #fff;
}

.nav-items {
  margin-right: 10px;
}

.header-logo {
  cursor: pointer;
  width: 20%;
  height: auto;
  position: absolute;
  left: -10px;
  top: 13px;
  opacity: 1;
}

#main-nav {
  margin: 0 20px 0 0;
}

/* the --main-color is a css variable found in index.css */
#main-nav a {
  color: var(--main-color);
}

#main-nav a:hover,
#main-nav li.active a {
  background-color: rgba(0, 0, 0, 0);
  color: var(--second-color);
}

@media(max-width: 768px) {
  .header-logo {
    height: 45px;
    top: 5px;
    left: 1px;
    width: auto;
  }
}

@media(min-width: 768px) {
  .nav-custom {
    padding: 20px 0;
    border-bottom: 0;
    letter-spacing: 1px;
    background: 0 0;
    transition: background .5s ease-in-out,padding .5s ease-in-out;
  }

  .header-logo {
    z-index: 1;
    height: 90%;
    width: auto;
    position: absolute;
    left: -10px;
    top: 5%;
    opacity: 1;
    transition: opacity 0.5s;
  }

  .nav-custom.top-nav-collapse {
    padding: 0;
    border-bottom: 1px solid rgba(0,0,0,.3);
    background: #fff;
  }
}