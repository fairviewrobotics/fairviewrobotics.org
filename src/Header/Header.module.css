.customNav {
  margin-bottom: 0;
  border-bottom: 1px solid rgba(0,0,0,.3);
  text-transform: uppercase;
  font-family: Montserrat, "Helvetica Neue", Helvetica, Arial, sans-serif;
  background-color: #fff;
}

.logo {
  cursor: pointer;
  width: 20%;
  height: auto;
  position: absolute;
  left: -10px;
  top: 13px;
  opacity: 1;
}

.nav {
  margin: 0 20px 0 0 !important;
}

/* the --main-color is a css variable found in index.css */
.nav a {
  color: var(--main-color) !important;
}

.nav a:hover,
.nav li:global(.active) a {
  background-color: rgba(0, 0, 0, 0) !important;
  color: var(--second-color) !important;
}

@media(max-width: 768px) {
  .logo {
    height: 45px;
    top: 5px;
    left: 1px;
    width: auto;
  }
}

@media(min-width: 768px) {
  .customNav {
    padding: 20px 0;
    border-bottom: 0;
    letter-spacing: 1px;
    background: 0 0;
    transition: background .5s ease-in-out,padding .5s ease-in-out;
  }

  .logo {
    z-index: 1;
    height: 90%;
    width: auto;
    position: absolute;
    left: -10px;
    top: 5%;
    opacity: 1;
    transition: opacity 0.5s;
  }

  .customNav.topNavCollapse {
    padding: 0;
    border-bottom: 1px solid rgba(0,0,0,.3);
    background: #fff;
  }
}