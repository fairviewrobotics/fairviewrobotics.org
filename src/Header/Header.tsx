import React, { Component } from 'react';
import { Navbar as BSNavBar, Nav, NavItem } from 'react-bootstrap';
import * as bootstrapUtils from "react-bootstrap/lib/utils/bootstrapUtils";
import { NavLink } from 'react-router-dom';
// This helps link 'react-router-dom' and 'react-bootstrap' together to make links work
import { LinkContainer } from "react-router-bootstrap";
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { faBars } from "@fortawesome/free-solid-svg-icons";

import styles from './Header.module.css';
import logo from '../images/logo.png';

export type HeaderProps = {
  isCollapsed: boolean;
}

// This remove the default styling from 'react-bootstrap' for the Navbar
bootstrapUtils.addStyle(BSNavBar, 'none');

class Header extends Component<HeaderProps> {

  render() {
    const collapsedStyling = this.props.isCollapsed ? styles.topNavCollapse : null;

    return (
      <BSNavBar collapseOnSelect fixedTop bsStyle="none" className={`${styles.customNav} ${collapsedStyling}`}>
        <BSNavBar.Header>
          <BSNavBar.Brand>
            <NavLink to="/">
              <img src={logo} alt="logo" className={styles.logo} />
            </NavLink>
          </BSNavBar.Brand>
          <BSNavBar.Toggle>
            <FontAwesomeIcon icon={faBars}/>
          </BSNavBar.Toggle>
        </BSNavBar.Header>
        <BSNavBar.Collapse>
          <Nav pullRight className={styles.nav}>
            <LinkContainer to="/about">
              <NavItem>About</NavItem>
            </LinkContainer>
            <LinkContainer to="/supporters">
              <NavItem>Supporters</NavItem>
            </LinkContainer>
            <LinkContainer to="/gallery">
              <NavItem>Gallery</NavItem>
            </LinkContainer>
            <LinkContainer to="/calendar">
              <NavItem>Calendar</NavItem>
            </LinkContainer>
            <LinkContainer to="/join-us">
              <NavItem>Join Us</NavItem>
            </LinkContainer>
          </Nav>
        </BSNavBar.Collapse>
      </BSNavBar>
    );
  }
}

export default Header;
