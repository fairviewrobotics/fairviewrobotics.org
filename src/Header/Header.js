import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { Navbar, Nav, NavItem } from 'react-bootstrap';
import * as bootstrapUtils from "react-bootstrap/lib/utils/bootstrapUtils";
import { NavLink } from 'react-router-dom';
// This helps link 'react-router-dom' and 'react-bootstrap' together to make links work
import { LinkContainer } from "react-router-bootstrap";
import FontAwesomeIcon from '@fortawesome/react-fontawesome';
import { faBars } from "@fortawesome/fontawesome-free-solid/shakable.es";

import './Header.css';

import logo from '../images/logo.png';

// This remove the default styling from 'react-bootstrap' for the Navbar
bootstrapUtils.addStyle(Navbar, 'none');

class Header extends Component {

  static propTypes = {
    isCollapsed: PropTypes.bool.isRequired
  };

  render() {
    const collapsedStyling = this.props.isCollapsed ? 'top-nav-collapse' : null;

    return (
      <Navbar collapseOnSelect fixedTop bsStyle="none" className={`nav-custom ${collapsedStyling}`}>
        <Navbar.Header>
          <Navbar.Brand>
            <NavLink to="/">
              <img src={logo} alt="logo" className="header-logo" />
            </NavLink>
          </Navbar.Brand>
          <Navbar.Toggle>
            <FontAwesomeIcon icon={faBars}/>
          </Navbar.Toggle>
        </Navbar.Header>
        <Navbar.Collapse>
          <Nav pullRight id="main-nav">
            <LinkContainer to="/about">
              <NavItem>About</NavItem>
            </LinkContainer>
            <LinkContainer to="/sponsors">
              <NavItem>Sponsors</NavItem>
            </LinkContainer>
            <LinkContainer to="/gallery">
              <NavItem>Gallery</NavItem>
            </LinkContainer>
            <LinkContainer to="/calendar">
              <NavItem>Calendar</NavItem>
            </LinkContainer>
          </Nav>
        </Navbar.Collapse>
      </Navbar>
    );
  }
}

export default Header;
