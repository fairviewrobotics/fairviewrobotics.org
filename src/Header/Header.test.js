import React from "react";
import { shallow } from "enzyme";
import Header from "./Header";
jest.mock('@fortawesome/fontawesome-free-solid', () => ({
  faBars: 'mockedfaBarsIcon'
}));

describe('Header', () => {
  let wrapper;

  it('renders without crashing', () => {
    shallow(
      <Header isCollapsed={false}/>
    );
  });

  it('renders a navbar with links and logo', () => {
    wrapper = shallow(
      <Header isCollapsed={false}/>
    );

    expect(wrapper).toMatchSnapshot();
  });

  describe('when it `isCollapsed`', () => {
    beforeEach(() => {
      wrapper = shallow(
        <Header isCollapsed={true}/>
      );
    });

    it('renders a collapsed navbar', () => {
      expect(wrapper).toHaveClassName('topNavCollapse');
    });
  });
});