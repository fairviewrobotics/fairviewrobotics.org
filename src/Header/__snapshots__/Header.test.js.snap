// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header renders a navbar with links and logo 1`] = `
<Uncontrolled(Navbar)
  bsStyle="none"
  className="customNav null"
  collapseOnSelect={true}
  fixedTop={true}
>
  <NavbarHeader>
    <NavbarBrand>
      <NavLink
        activeClassName="active"
        ariaCurrent="true"
        to="/"
      >
        <img
          alt="logo"
          className="logo"
          src="logo.png"
        />
      </NavLink>
    </NavbarBrand>
    <NavbarToggle>
      <FontAwesomeIcon
        border={false}
        className=""
        fixedWidth={false}
        flip={null}
        icon="mockedfaBarsIcon"
        listItem={false}
        mask={null}
        name=""
        pull={null}
        pulse={false}
        rotation={null}
        size={null}
        spin={false}
        symbol={false}
        transform={null}
      />
    </NavbarToggle>
  </NavbarHeader>
  <NavbarCollapse>
    <Nav
      bsClass="nav"
      className="nav"
      justified={false}
      pullLeft={false}
      pullRight={true}
      stacked={false}
    >
      <LinkContainer
        activeClassName="active"
        exact={false}
        replace={false}
        strict={false}
        to="/about"
      >
        <NavItem
          active={false}
          disabled={false}
        >
          About
        </NavItem>
      </LinkContainer>
      <LinkContainer
        activeClassName="active"
        exact={false}
        replace={false}
        strict={false}
        to="/sponsors"
      >
        <NavItem
          active={false}
          disabled={false}
        >
          Sponsors
        </NavItem>
      </LinkContainer>
      <LinkContainer
        activeClassName="active"
        exact={false}
        replace={false}
        strict={false}
        to="/gallery"
      >
        <NavItem
          active={false}
          disabled={false}
        >
          Gallery
        </NavItem>
      </LinkContainer>
      <LinkContainer
        activeClassName="active"
        exact={false}
        replace={false}
        strict={false}
        to="/calendar"
      >
        <NavItem
          active={false}
          disabled={false}
        >
          Calendar
        </NavItem>
      </LinkContainer>
    </Nav>
  </NavbarCollapse>
</Uncontrolled(Navbar)>
`;
